src/Momento.Sdk.Incubating/Responses/CacheDictionaryGetFieldsResponse.cs

```diff
@@ -22,20 +22,23 @@ public class Hit : CacheDictionaryGetFieldsResponse
         public Hit(IEnumerable<ByteString> fields, _DictionaryGetResponse responses)
         {
             var responsesList = new List<CacheDictionaryGetFieldResponse>();
+            List<ByteString> fieldsList = fields.ToList();
+            var counter = 0;
             foreach (_DictionaryGetResponsePart response in responses.Found.Items)
             {
                 if (response.Result == ECacheResult.Hit)
                 {
-                    responsesList.Add(new CacheDictionaryGetFieldResponse.Hit(response.CacheBody));
+                    responsesList.Add(new CacheDictionaryGetFieldResponse.Hit(fieldsList[counter], response.CacheBody));
                 }
                 else if (response.Result == ECacheResult.Miss)
                 {
-                    responsesList.Add(new CacheDictionaryGetFieldResponse.Miss());
+                    responsesList.Add(new CacheDictionaryGetFieldResponse.Miss(fieldsList[counter]));
                 }
                 else
                 {
-                    responsesList.Add(new CacheDictionaryGetFieldResponse.Error(new UnknownException(response.Result.ToString())));
+                    responsesList.Add(new CacheDictionaryGetFieldResponse.Error(fieldsList[counter], new UnknownException(response.Result.ToString())));
                 }
+                counter++;
```

Comment:
just for a learning opportunity, check out this: https://thomaslevesque.com/2019/11/18/using-foreach-with-index-in-c/

it shows an example of how you might be able to loop with a counter/index var without having to define the local mutable variable.

I am not familiar enough with c# to look at the code in that example and determine whether the `Select` operation is lazy or not; if it's not, then the code in that blog would change the cost of this method from `O(n)` to `O(2n)`, so I think it's fine for you to just leave this how it is, but I wanted to share with you what I was thinking about when I looked at it.

Reply:
Thank you for the advice 🙇‍♀️ 
I'm not 100% sure if I'm looking at the right place in the source code but it looks like the `Select` operations also uses `foreach` under the hood. 
https://referencesource.microsoft.com/#System.Core/System/Linq/Enumerable.cs,53
So if that's correct, as you mentioned, the blog's example's time complexity would be `O(2n)` instead of `O(n)`.

Using the local mutable variable is not preferred?

Reply:
I always try to avoid mutable variables any time it is possible to do so.  Eliminating them reduces the possibility of all kinds of bugs.

But in this case if it will increase the time complexity by a factor of 2 then sticking with the mutable variable is okay.  A lot of languages would have a function like `forEachWithIndex` for this, which would avoid the mutable variable and also still have the `O(n)` time complexity.  But it looks like C# doesn't have that.

If you are bored and interested in more background on my thoughts on mutable vars you can check out this blog post that I wrote sometime :)  https://www.gomomento.com/blog/moving-your-bugs-forward-in-time

Reply:
I see. It makes sense.
Yes! I will read the blog! (I was actually going to go through some of our blogs since they are so informational 😄 )